### ISSUE TITLE
Unigram 9.5.8088.0 crashes on Windows on ARM

### ISSUE THREAD

**Issue by AkinoKaede** [UnigramDev/Unigram#3010]
**Describe the bug**
Unigram 9.5 crashes on Windows on ARM, although I reinstall it.

**To Reproduce**
1. Install Unigram in Microsoft Store
2. Open it

**Expected behavior**

**Screenshots**

**Version Info**
 - Unigram: 9.5.8088.0
 - Windows: Windows 11 22H2 22621.1485

**Additional context**
CPU: Qualcomm Snapdragon 8cx Gen 3

Sorry, I don't know how to get the crash log, could anyone tell me?

**Comment by punteroanull** [comment]
Same here (Surface Pro 9 5G),  I installed the official version meanwhile.

**Comment by MarcAnt01** [comment]
Actually I think it's too slow. If the app is taking so much I prefer it to delay the splash screen rather than showing the app without the main content

https://user-images.githubusercontent.com/43090263/228591212-9dc07f0d-70ff-4bbf-819e-0adc3fcef441.mp4

**Comment by MarcAnt01** [comment]
Sorry for the message, it was meant for another repository

**Comment by vitaminj** [comment]
Same, I get the splash and then the window immediately closes (Volterra Dev Kit)

**Comment by MarcAnt01** [comment]
Hi everyone thanks for reporting immediately, we are aware of this issue and its cause. A fix is being worked on.

@punteroanull TD has no Arm version, you simply emulate the x64/x86 version, being like this it requires no mantainance from the developers, since Unigram has an Arm version it requires some mantainance and this kind of issues can occur, we are sorry for the inconvenience of having to use another software temporarily

**Comment by MarcAnt01** [comment]
Fixed in 8.5.8101

**Comment by AkinoKaede** [comment]
> Fixed in 8.5.8101

I upgrade it to 9.5.8106，but the problem doesn't seem to be resolved.

**Comment by FrayxRulez** [comment]
Can someone provide a crash dump?

**Comment by domcote** [comment]
Happy to. Can you pls provide some instructions for noobs? 😁

**Comment by punteroanull** [comment]
> Can someone provide a crash dump?

Nombre de registro:Application
Origen:        Windows Error Reporting
Fecha:         04/04/2023 18:37:10
Id. del evento:1001
Categoría de la tarea:Ninguno
Nivel:         Información
Palabras clave:
Usuario:       SURFACEDU\eruiz
Equipo:        SurfacEdu
Descripción:
Depósito con errores , tipo 0
Nombre de evento: POFContextAppCrash
Respuesta: No disponible
Identificador de archivo .cab: 0

Firma del problema:
P1: 38833FF26BA1D.UnigramPreview_9.5.8106.0_arm64__g9c9v27vpyspw
P2: praid:App
P3: 9.5.8106.0
P4: 0x80040154
P5: 69463437391835439dc7c5a67896be75
P6: 
P7: 
P8: 
P9: 
P10: 

Archivos adjuntos:

Es posible que estos archivos estén disponibles aquí:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_38833FF26BA1D.Un_7d1569ff2373de29bfcb1a677d72715ee7f33_00000000_5c0b0481-ad78-4ca3-8f45-a813bfa55c61

Símbolo de análisis: 
Nueva búsqueda de una solución: 0
Id. de informe: 5c0b0481-ad78-4ca3-8f45-a813bfa55c61
Estado del informe: 4
Depósito con algoritmo hash: 
GUID de archivo .cab: 0
XML de evento:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Windows Error Reporting" Guid="{0ead09bd-2157-539a-8d6d-c87f95b64d70}" />
    <EventID>1001</EventID>
    <Version>0</Version>
    <Level>4</Level>
    <Task>0</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000000</Keywords>
    <TimeCreated SystemTime="2023-04-04T16:37:10.3319009Z" />
    <EventRecordID>18468</EventRecordID>
    <Correlation />
    <Execution ProcessID="25628" ThreadID="24928" />
    <Channel>Application</Channel>
    <Computer>SurfacEdu</Computer>
    <Security UserID="S-1-5-21-2817870534-2582539154-1409932555-1003" />
  </System>
  <EventData>
    <Data Name="Bucket">
    </Data>
    <Data Name="BucketType">0</Data>
    <Data Name="EventName">POFContextAppCrash</Data>
    <Data Name="Response">No disponible</Data>
    <Data Name="CabId">0</Data>
    <Data Name="P1">38833FF26BA1D.UnigramPreview_9.5.8106.0_arm64__g9c9v27vpyspw</Data>
    <Data Name="P2">praid:App</Data>
    <Data Name="P3">9.5.8106.0</Data>
    <Data Name="P4">0x80040154</Data>
    <Data Name="P5">69463437391835439dc7c5a67896be75</Data>
    <Data Name="P6">
    </Data>
    <Data Name="P7">
    </Data>
    <Data Name="P8">
    </Data>
    <Data Name="P9">
    </Data>
    <Data Name="P10">
    </Data>
    <Data Name="AttachedFiles">
    </Data>
    <Data Name="StorePath">\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_38833FF26BA1D.Un_7d1569ff2373de29bfcb1a677d72715ee7f33_00000000_5c0b0481-ad78-4ca3-8f45-a813bfa55c61</Data>
    <Data Name="AnalysisSymbol">
    </Data>
    <Data Name="Rechecking">0</Data>
    <Data Name="ReportId">5c0b0481-ad78-4ca3-8f45-a813bfa55c61</Data>
    <Data Name="ReportStatus">4</Data>
    <Data Name="HashedBucket">
    </Data>
    <Data Name="CabGuid">0</Data>
  </EventData>
</Event>

**Comment by FrayxRulez** [comment]
Grabbed one from my Mac mini. Ironically, Visual Studio doesn't support inspecting stowed exceptions from ARM64 mini dumps.

**Comment by MarcAnt01** [comment]
Hi Arm64 users, just a quick update. In order not to prevent you from using Unigram the Arm64 has been removed from the store, so that you can keep using Unigram (even if in x64/x86), it's planned to fix the crash so that the Arm version can be sent back to the store in the future

**Comment by bam2000** [comment]
any update on this?

**Comment by FrayxRulez** [comment]
There are currently no plans to bring back support to ARM64. 
Given the meager amount of users and lack of support by multiple app dependencies, this is not worth the effort.

I'm not closing this, as there's always the hope that things will change in the future.

**Comment by Soneoylys** [comment]
> There are currently no plans to bring back support to ARM64. Given the meager amount of users and lack of support by multiple app dependencies, this is not worth the effort.
> 
> I'm not closing this, as there's always the hope that things will change in the future.

I have a Snapdragon 860 tablet with Windows 11 installed, if I want to compile the old known work version for myself as I'm not bothering to use a slightly old version, may I using cross-compile on amd64 desktop?
I'm able to install Visual Studio on the tablet but due to lack of ram (which only have 6 gigs) it's hard for compile on itself, or is there any old prebuild installer archive which still support arm64 can download?

**Comment by FrayxRulez** [comment]
@Soneoylys Yes. It is possible to cross-compile, still compiling Unigram is not an easy task.

**Comment by Soneoylys** [comment]
> @Soneoylys Yes. It is possible to cross-compile, still compiling Unigram is not an easy task.

Thanks! I've succeeded cross-compiled arm64 build.

To someone wants to try: Confirmed [this release](https://github.com/UnigramDev/Unigram/releases/tag/v8.3) works, follow the build instructions in [this file](https://github.com/UnigramDev/Unigram/blob/7e900c82edbc92d58725b26f5e160453adb66828/Documentation/Build-instructions.md), but while you configure the Visual Studio, make sure you install Windows 10 SDK as well, otherwise vcpkg might throw an exception about cannot found SDKs.

**Comment by domcote** [comment]
NICE!

thanks for working on this. 👍

Do your have binaries I can test? I'm no dev and compiling stuff is beyond me. 😁

Dom
________________________________
Von: Soneoylys ***@***.***>
Gesendet: Mittwoch, 22. November 2023 20:14
An: UnigramDev/Unigram ***@***.***>
Cc: domcote ***@***.***>; Comment ***@***.***>
Betreff: Re: [UnigramDev/Unigram] Unigram 9.5.8088.0 crashes on Windows on ARM (Issue #3010)


@Soneoylys<https://github.com/Soneoylys> Yes. It is possible to cross-compile, still compiling Unigram is not an easy task.

Thanks! I've succeeded cross-compiled arm64 build.

To someone wants to try: Confirmed this release<https://github.com/UnigramDev/Unigram/releases/tag/v8.3> works, follow the build instructions in this file<https://github.com/UnigramDev/Unigram/blob/7e900c82edbc92d58725b26f5e160453adb66828/Documentation/Build-instructions.md>, but while you configure the Visual Studio, make sure you install Windows 10 SDK as well, otherwise vcpkg might throw an exception about cannot found SDKs.

**Comment by Soneoylys** [comment]
> NICE! thanks for working on this. 👍 Do your have binaries I can test? I'm no dev and compiling stuff is beyond me. 😁 Dom

The app bundle's certification signature will not valid cross device.

Technically you can trust a custom ROOT CA to workaround AFAIK, but it also could be dangerous if you install CA from unknown person, which can do A LOT including decrypt your HTTPS data.

So I don't really want to upload precompiled binary with my custom signed CA because of that, sorry.

**Comment by ofoacimr** [comment]
> There are currently no plans to bring back support to ARM64. Given the meager amount of users and lack of support by multiple app dependencies, this is not worth the effort.
> 
> I'm not closing this, as there's always the hope that things will change in the future.

Hi, I'm wondering if there are any chances to use Arm64EC? It can contain x64 binary and x64 lib.

> By recompiling the modules that take the most time or are the most CPU-intensive from x64 to Arm64EC, the resulting workload receives the most improvement for the least amount of effort each step of the way.

https://learn.microsoft.com/en-us/windows/arm/arm64ec

**Comment by gus33000** [comment]
The developer of this project repeatedly claims as part of the reason for not bothering to port the app to arm64 that the userbase is too low to care about this.

I would be interested in the source for above claim as well as how it was gathered (telemetry for an app compiled as x64 can only report x64 obviously)

With the quick emergence of even more hw, it is also irresponsible to continue denying the eventual possibility of looking into doing such a port.

I also seriously wonder if it's really about business justification on userbase, or just personal hate?

To cite the developer: (message sent on telegram publicly once) 

```
In my opinion they can suffer. Maybe it’s Microsoft choice not to provide a better emulation nor to push properly for arm64 adoption as Apple started to do 4 years ago.
```

**Comment by FrayxRulez** [comment]
> I would be interested in the source for above claim as well as how it was gathered

The app has natively supported ARM64 until recently (Can't say exactly when, but I would say late 2022, early 2023) and it was one of the first (widely used) UWP apps to do that.

As the ARM64 user base has always been insignificant (we're talking about a few thousands people on the peak) I've decided to drop the support, because it's in fact non trivial, as Unigram has few native external dependencies that just don't care about ARM64 support on Windows (while they do on UNIX/Darwin) and they require writing patches and fixes for them and their build scripts every time they change something.

While I did this for years, I decided that's not worth it given the meaningless impact on user base.

Regarding Arm64EC comment from @ofoacimr, I don't think that this is doable: main code base for Unigram is C# compiled using .NET Native toolchain, and I'm not aware of Arm64EC for .NET Native.

**Comment by FrayxRulez** [comment]
By the way, I think it's a bit unfair to post that quote completely decontextualized from the rest of the conversation:
![image](https://github.com/UnigramDev/Unigram/assets/8152849/df485814-2284-42a3-ab25-eb9a1fe21feb)

**Comment by vitaminj** [comment]
Thankyou for your previous support (and considered response here), Unigram was one of the few when I first bought my Volterra box that I was pleasantly surprised had an ARM version!
It's a shame the deps broke, but thankfully Unigram isn't exactly a CPU hog that needs to be run natively, so emulation is a perfectly fine fallback (and not noticeable for me).
Let's hope that the dependencies change their stance now that all the Snapdragon X stuff is coming and there is a legitimate push (after the small push with Volterra which kinda fizzled out)

**Comment by FrayxRulez** [comment]
@vitaminj agreed! Let's see what happens. Of course, it there will be a growth I'll be happy to get those libraries compatible again by myself (but hopefully it won't be needed)

**Comment by gus33000** [comment]
> By the way, I think it's a bit unfair to post that quote completely decontextualized from the rest of the conversation:
> ![image](https://github.com/UnigramDev/Unigram/assets/8152849/df485814-2284-42a3-ab25-eb9a1fe21feb)
> 

In my honest opinion I don't see how the context you added here changes the messaging in a significant way.

I appreciate your reply but I'll remain on a disagreement with you on marketshare that makes little to no point. It really feels more like you don't want to care about the platform which is a real shame.

Calling my comment on this thread childish on different channel does nothing but recomfort me in the feeling that you really don't want to provide the right argument into this matter. While I appreciate your answer and obviously understand you're working alone on this app (and am thankful as well for this), I among others have also wanted to help you make that switch, but due to incredibly difficult instructions to build this app on its intended target at the moment (x64) and past issues reproducing the build steps, (for various reason, unpushed changes hard-coded paths etc), I at least never followed to contribute to make this switch as it simply feels unwelcoming and takes too much of our time to help. This may have changed now, I haven't checked this year. But I hope you at least understand the mood feels a bit unwelcome here, especially when everytime the conversation or state moves elsewhere. At the same time, it might make more sense to contribute into different project firsts. Tdesktop marketshare is what compared to unigram?

**Comment by gus33000** [comment]
In my honest opinion the real childish move here is the way you reply to this. I wasn't after all being in the wrong. Banning users because they offer help for you and want to make you change your stance and realizing supporting an emerging platform is the real childish thing. I'm thankful at least I got to know your real reason here. Unless you really want to recomfort us and excuse you for this behavior.

**Comment by FrayxRulez** [comment]
> In my honest opinion I don't see how the context you added here changes the messaging in a significant way.

A clearly provocative message gets replied accordingly.
Talking about "making someone suffer" because they're forced to use a software compiled using an architecture that's not their native one is far from making someone suffer.

If you feel discouraged by Unigram build instructions and you don't want to contribute, what does it make you think that's any different for me towards all the libraries that don't support ARM64 on MSVC by so many years and they care absolutely nothing about it? 

Chrome rolled out ARM64 support just a couple of months ago, I guess that now it should be fairly easy to compile WebRTC for the architecture. Until then, WebRTC itself and 4-5 of its dependencies required patches to compile on ARM64.

WebRTC still requires a [10.000 lines patch](https://github.com/FrayxRulez/webrtc-uwp/commit/74b8a0e6feda5a721a1c5c4b93fbd313f2dc0d91) to work with UWP, and this as well requires a lot of maintenance on every release.

LibVLC is not shipped pre-built for ARM64 and I'm not sure whether compiling it for UWP ARM64 is possible at all, because their community is as welcoming as this one and their documentation is just terrible and broken (but still, I did my best at least to compile their library for x64 by myself).

FFmpeg somehow builds on ARM64 but decoded videos are full of artifacts and glitches. Still, depending on the version it won't build or would just crash on x64 because of some bug in their ASM code that will likely never get fixed because their community is as helpful as this one (by "this one", I of course refer to myself), so we're stuck on some ancient version that at least won't crash.

I can keep on going, if you want.

This said, I find it extremely offensive that you consider your time so much valuable not to want to invest any of it in trying to do something (that currently) would be helpful only for you and a few more people, but you decide to ignore all of the above and you accuse me of "hating" and "not caring".

**Comment by gus33000** [comment]
I don't see how calling my message childish and every time the arm topic is brought up saying the marketshare is low makes me any welcome to help here? If you find the message you said yourself offensive, so do I? I wasn't intending the quote to be provocative, it more highlights one of the few examples where it felt like you preferred us to suffer other than having a native build. I of course remain happy to help one day, but only if we can have a conversation on the topic other than calling each other childish and if you're willing to help us a little as well. Saying every user should suffer is a bit wrong. The emulation layer isn't even originally this bad. People benchmarked it on apple laptops originally where it was as good as rosetta

It wouldn't even benefit just me, there's a lot more users, plus, you use an arm MacBook for a while, you had access to this architecture no? Unless I'm mistaken and I am sorry if I am.

I don't see how I'm supposed to feel welcome or happy to help here, you maybe still have a point on marketshare if you want to but at the same time if we really have to consider marketshare it would make way more sense to port tdesktop first. I really am not here motivated by marketshare. I just wish the in my personal opinion, best telegram experience for windows users would be native again, and I would be happy to help. But you really don't encourage me to continue.

Just today users were discussing using telegram web as a workaround. I wish they would have a native version of this app. It's not just me thinking you don't care here. If I'm wrong, I'm sorry and I would be happy to be. But others say it too. And I really am not sure the marketshare is this low to begin. With...

Happy to forget what happened but only if you are willing. 

I dont think asking you to at least, make the ability to build the app on amd64 easier from scratch or at least helping us doing so is a hard feat for you, you do this all the time to build the app anyway, its what really doesnt help us at all trying to contribute. Why would I spend time myself guessing how you build the app normally here, when you could just help us do it? Is it that wrong to not want to spend the time doing that guesswork? And as you said, it wouldnt even be just, here have premade libs, we obviously need to know how you built them...

so no, of course i dont want to help here, you ban us, react in a negative way, and are often unwilling to help us rebuild the app from scratch. How does that make me the one being in the wrong for not wanting to?

**Comment by GeldHades27355** [comment]
Foks, love the discussion right now.

Why don't we wait a few weeks/months and revisit this? Especially @FrayxRulez I bet with the advent of Snapdragon X for Windows and MSFT's _extreme_ interest in establishing it as the NEXT benchmark for Windows experiences, there will be a flurry of new development and compilation, including all the Unigram dependencies.

Going forward, devs and their business manager can't afford to ignore Arm64 support anymore, as even large hardware and software corporations are fully buying in to Arm64 now.

As pointed out, Unigram runs quite well as an emulated x64 app right now, so there shouldn't be real urgency. But I see how other universal apps have gotten **super-charged** after they switched from emulated x64 to native Arm64 code.

@FrayxRulez while we're at it, can you ensure there are no Arm32 fragments left? Windows will drop support for Arm32 entirely by 24H2. Current previews already won't run Arm32 anymore (Minecraft now is emulated... 😭)

**Comment by FrayxRulez** [comment]
> Why don't we wait a few weeks/months and revisit this?

Of course, I never said never, I always just referred to _right now_.

**Comment by gus33000** [comment]
What tells us you're going to test these builds later on? Afterall, twice in the past they broke, you referred before as 2022, [but you forgot the app broke the year before](https://github.com/UnigramDev/Unigram/issues/2645#issuecomment-1040300071). Even if I'm willing to help you kickstart the port here, what tells me everything will have to be redone later on because the app broke with some update due to a lack of testing, and that it's going to come back to x64 only? Or that you adopt a new library one day (like libVLC which didn't use to exist) We really need this assurance as well.

I fully get you want to talk about right now and that you'll revisit this, but the history really isn't helping.

The argument about chrome itself just being out is also awkward. Chromium has been running on arm64 for years, and that has always used WebRTC. I in fact made WebRTC compile for you years ago... (remember: https://gist.github.com/gus33000/7a7f310528082d5b7ba555fed6379e03)

And I'm really not taking the argument well that I'm not even contributing here on my own before complaining, who made a guide to help you build your own app myself, spent an entire two weeks figuring out what exact commit hashes you even used to build it (which again, today, proves to be likely the reason as to why people can't rebuild it, a few examples: https://github.com/UnigramDev/Unigram/issues/3095 https://github.com/UnigramDev/Unigram/issues/3038 or this one, where you simply told the person to go fix it on his own... https://github.com/UnigramDev/Unigram/issues/3068), and correct the hardcoded paths? Like i'm sorry, but i find the argument very unjustified.

https://gist.github.com/gus33000/7a7f310528082d5b7ba555fed6379e03

To me the app doesn't simply need to wait on some library maintainers to fix their things (and i agree with you here), it also needs assurance that you're going to commit to test and maintain this platform afterall. Can we trust you on this? Or would you want to rely on others for doing so?

The problem is bigger when the app suddenly fails one day compared to the previous one, than when it's not pefoming as it should.

Once again, I'm more than willing to provide feedback and help here, but only if you don't act in a hostile way to us and your end users that only realistically complain because they love your app and just want it to run great on their device.

**Comment by FrayxRulez** [comment]
> The argument about chrome itself just being out is also awkward. Chromium has been running on arm64 for years

As on the latest version of WebRTC used by Telegram (m123) ARM64 fails to compile due to some assembly in dav1d.

> To me the app doesn't simply need to wait on some library maintainers to fix their things (and i agree with you here), it also needs assurance that you're going to commit to test and maintain this platform afterall. Can we trust you on this? Or would you want to rely on others for doing so?

I have never tried Unigram on ARM64, nor on x86 or Xbox.
I can't guarantee anything, especially until ARM64 user base is neglectable.

**Comment by gus33000** [comment]
> I have never tried Unigram on ARM64, nor on x86 or Xbox. I can't guarantee anything, especially until ARM64 user base is neglectable.

You either have to test it yourself on your own macbook, or rely on the "small" userbase that exists. 

Banning us or telling every single one of us that you don't want to currently port the app really is not going to help giving you any feedback on this, and you might as well openly say you will not want to care about this platform. You disagree with me above when I said you didn't care, but that's exactly what you're saying above. Of course! that's wrong if you really value the feedback the already small userbase sends you, but how do you think i take the action you did today towards me over something I haven't even said in your feedback channel? (and it wasn't even hostile)

How do you think the already small userbase of xbox users will react to you saying you have never tried to see if the app worked well on the device they use? And i can conceive not having an xbox, but why supporting it then? Same for x86? Or is the app really not supported for anyone?

You either care or you don't, but it has to be clear. Having the app work one day on ARM64, just like it launching on Xbox, is not going to be really enough, there's a need for testing, either directly from you (on all sorts of hw) or otherwise from others. Otherwise, it really is best to **never support Windows on ARM**.

**Comment by shijunz** [comment]
> dav1d.

do you try to change the clang-cl?

call gn gen --ide=vs2022 out\msvc\uwp\%%c\%%a --filters=//:webrtc "--args=is_debug=!is_debug! use_lld=false is_clang=true rtc_include_tests=false rtc_build_tools=false rtc_win_video_capture_winrt=true target_os=\"winuwp\" rtc_build_examples=false rtc_win_use_mf_h264=true rtc_use_h265=true rtc_enable_protobuf=false rtc_disable_metrics=true rtc_include_dav1d_in_internal_decoder_factory=false treat_warnings_as_errors=false use_custom_libcxx=false fatal_linker_warnings=false target_cpu=\"%%a\""



is_debug = false
use_lld = false
is_clang = false  // need change true

maybe the clang-cl is supported and fix the assembly  issue.

**Comment by GeldHades27355** [comment]
Folks, i dunno but it looks like we're WAY OT.

I can say that Unigram has been emulating nicely on our Snapdragons for a very long time now. MSFT really seems to have the x86/x64 emulation nailed. 💪
It even seems quite speedy and efficient - battery usage is extremely low.

The only annoying thing right now is that video calls have nearly unbearable latency, but it seems to be the same on mobile. So I couldn't pin this on Unigram, seems more like an issue with the platform all up.

Should we close this issue and shift any Arm64 enablement/migration to another more appropriate thread?

**Comment by asdetycv1zzc** [comment]
I have already somehow resolved the VLC UWP build & TDLib build for native ARM64 by really lots of tricky workarounds. But the building of WebRTC blocked me from further continuing my work.
![image](https://github.com/user-attachments/assets/1c8d81c2-1e02-4983-957e-8700e324740c)
Maybe it is because my environment variables are broken? Or just the git repo had changed while acquire.cmd left unmodified?

**Comment by asdetycv1zzc** [comment]
> Folks, i dunno but it looks like we're WAY OT.
> 
> I can say that Unigram has been emulating nicely on our Snapdragons for a very long time now. MSFT really seems to have the x86/x64 emulation nailed. 💪 It even seems quite speedy and efficient - battery usage is extremely low.
> 
> The only annoying thing right now is that video calls have nearly unbearable latency, but it seems to be the same on mobile. So I couldn't pin this on Unigram, seems more like an issue with the platform all up.
> 
> Should we close this issue and shift any Arm64 enablement/migration to another more appropriate thread?

Well yes, its emulation runs smoothly and has a nice performance.
However it sometimes crashes my Touch Screen Driver for unknown reasons so it will still be better for me if we can have a native ARM64 version.

**Comment by FrayxRulez** [comment]
@asdetycv1zzc Qualcomm is currently working on getting the app to run natively on ARM64 so I think we can wait for their input

**Comment by asdetycv1zzc** [comment]
After a lot of work I realized that the Doc of building may have gone incompatible with ARM64. Many packages like vcpkg and vlc needs to be in the latest version to make the building process of ARM64 possible.

For examle, I use the latest vcpkg instead of a specific checkout and succeeded, while the specific version of vcpkg mentioned in the tutor failed to install the essentials like zlib:arm64-uwp. VLC could be built by the old docker image but it failed to pack into exe (that's weird, why it attempted to output .exe files?) and lack libopus support (I just simply cannot compile libopus in ARM64 by the 2020 llvm image).

And also I found that repo WebRTC-UWP seems to be out of maintainence, for its last update was 3 yrs ago. So here comes the question that, which version of the packages that Unigram relies on should I choose, the older but tested ones or the newer ones which may support ARM64 native build w/o hacking & patches? I am totally confused now.

**Comment by FrayxRulez** [comment]
This is more complicated than it may look.
The version of vcpkg is quite important, because with newer versions there are some issues with ffmpeg on x64 (namely, pixel conversions from YUV to RGB may cause access violations and memory corruptions because of a bug that causes the conversion to write outside of the target buffer), so it's quite necessary to stay on this version.

VLC is honestly a mess and I regret having migrated to it, as the project feels completely abandoned in a way and even building x64 was a nightmare because of expired caches on their side and what not.

Unigram doesn't use WebRTC-UWP but my own fork (based on a old port from Microsoft for WinRT and updated to M123, that's the version of WebRTC used by Telegram.
I managed to build it for ARM64 in the past (as well as all the vcpkg dependencies) so it should be quite trivial to, although it be needed to change some of the library code/build scripts.

The build scripts for WebRTC may not be correct because I have never used them myself...
The idea is that you have to apply some patches that are included in this repository and they should apply properly if you checked out to M123 branch (that's my fork's branch).

**Comment by asdetycv1zzc** [comment]
Indeed I met some obvious problems like I cannot checkout m123 branch before pulling from upstream lol (.

Another major problem is that by using the past vcpkg I cannot even find arm64-triplet toolchains while the lastest vcpkg can detect it correctly. As I said, it is just confusing.

M123 branch, as the above discussions mentioned, has problems of ARM64 assembly files which ruins the building process. Does reversing it to a past version may work?

The last problem is that how can I build Unigram w/o WebRTC support? I followed the steps in the instructions and it does not work at all (it displays that many modules still referenced the files though I have removed "ENABLE_CALLS" macro)

**Comment by FrayxRulez** [comment]
> The last problem is that how can I build Unigram w/o WebRTC support? I followed the steps in the instructions and it does not work at all (it displays that many modules still referenced the files though I have removed "ENABLE_CALLS" macro)

Calls have been recently rewritten and the new code does not support `ENABLE_CALLS` flag. I'll try to reimplement it when I have time.

Anyhow as I mentioned Qualcomm is already working on porting the app to ARM64 and as from last report they got the app running, so we're just waiting for their pull request.

**Comment by Yvmsi1100** [comment]
****

**Comment by FrayxRulez** [comment]
Done

**Comment by gus33000** [comment]
Is it finally available in the store? If not do you have a rough estimate? Awesome either way thank you!

**Comment by FrayxRulez** [comment]
It will be available in the Store as soon as Microsoft approves it, so I guess early next week.

**Comment by gus33000** [comment]
I see thanks! The usual burden of the store........ ```</sigh>```

**Comment by GeldHades27355** [comment]
Just got the Arm64 version through the store. And what can I say... WOW. 🔥🔥 It is FAST. ⏩⏩⏩

It starts up near instantly and is ULTRA responsive on my SQ3 Surface Pro 9. In fact, it is snappier than than x64 on a core i5 I have lying around.

No crashes or issues yet, GIFs work, audio playback starts instantly, emojis (win+.) work, stickers work.

It currently consumes ~138MB of RAM, not sure if that is in line with expectations (I forgot to compare with emulated x64 code).

For now: AWESOME JOB TEAM! 👍💪

Please extend compliments to the folks at Qualcomm for their support as well.

**Comment by TGSAN** [comment]
Finally!!! I almost thought ARM64 was forgotten. Thanks to Qualcomm and all the contributors for their efforts.

**Comment by punteroanull** [comment]
Thank you all, guys, amazing job!

**Comment by Alovchin91** [comment]
Support for ARM64 has been re-added and retracted endless amount of times. I wouldn't want to update Unigram one day and receive an x64 version again. @FrayxRulez How reliable are these contributions from Qualcomm? Will they keep supporting Unigram? Do they make it less of a burden for you?

**Comment by GeldHades27355** [comment]
> Support for ARM64 has been re-added and retracted endless amount of times. I wouldn't want to update Unigram one day and receive an x64 version again. @FrayxRulez How reliable are these contributions from Qualcomm? Will they keep supporting Unigram? Do they make it less of a burden for you?

The Store does that all automatically and transparently. Both the switch from the old Arm32 universal to the x64 version and then to the new Arm64 version happened without me noticing. Except that the new Arm64 version started instantly this Monday while the emulated version is very noticeably slower.

So unless you're not using the Store, you don't need to worry about or deal with this.

**Comment by Alovchin91** [comment]
Well that's the reason why I'm asking: I don't want it to be automatically updated to x64 one day.

If it's not reliable, I'd rather stick with the official Telegram app where I suppose I can control the update process (and which I presume won't be dropping the Arm64 version randomly).

**Comment by FrayxRulez** [comment]
@Alovchin91 I think that given the current market share (arm64 users have been floating around 2.000 people during the last 6 months and this number doesn’t seem to be growing anyhow) it’s really hard to make any promise for both Unigram and Telegram Desktop. 
I think it’s quite important to underline here that the problem with arm64 is not the complexity of maintaining the compatibility while it’s rather the time vs impact. Build times double, it becomes more complicated to update dependencies, testing requires dedicated hardware, we have some build machines that can’t compile arm64, and the list keeps going. All of this for something that at its peak it is used by less than 0.1% of the users. 
Now I understand and I appreciate the enthusiasm and I don’t see why I should be dropping arm64 after all, but I’d still ask you to put yourself in my shoes and try to understand the situation. Also I don’t really appreciate being blackmailed (because this is what your last messages feel)

**Comment by Alovchin91** [comment]
Please don't make up something that's not there.

You've shared numerous times that it's a burden for you. I tried to be compassionate, so my question was — is it still a burden? From your response I read that it still is.

Unfortunately, every time I asked you about Arm64 you seemed to get upset, and your "blackmail" sentiment is exactly what creates that feeling once again.

So I simply wanted to explain why I'm asking again (because I just want to choose the client for myself, and while I really like Unigram, I don't want an auto-update to x64). I think it's a fair thing to share.

So I suppose I'll just leave you alone and I'm sorry if I upset you again.

Thank you for the response and your work.

**Comment by GeldHades27355** [comment]
> Please don't make up something that's not there.
> 
> You've shared numerous times that it's a burden for you. I tried to be compassionate, so my question was — is it still a burden? From your response I read that it still is.
> 
> Unfortunately, every time I asked you about Arm64 you seemed to get upset, and your "blackmail" sentiment is exactly what creates that feeling yet again.
> 
> So I simply wanted to explain why I'm asking again (because I just want to choose the client for myself, and while I really like Unigram, I don't want an auto-update to x64). I think it's a fair thing to share.
> 
> So I suppose I'll just leave you alone and I'm sorry if I upset you again.

The official Telegram client is x64 ONLY. Unigram seems to be the only client that is native Arm64 right now.
Abandoning Unigram does not solve the x64 question.

If you prefer Arm64, like the rest of us, you HAVE to use Unigram. Just do it, you have nothing to lose. Worst case: Arm64 is dropped again in a few months. Until then, you get the best experience.

**Comment by Alovchin91** [comment]
> The official Telegram client is x64 ONLY.

Not anymore (you'll find some links in this thread) but I'd like to refrain from discussing this since I don't want to be labelled a "blackmailer" anymore.

**Comment by GeldHades27355** [comment]
> > The official Telegram client is x64 ONLY.
> 
> Not anymore (you'll find some links in this thread) but I'd like to refrain from discussing this since I don't want to be labelled a "blackmailer" anymore.

Would you mind sharing where to get that? Neither the website nor the Store offer an Arm64 version of Telegram.

**Comment by FrayxRulez** [comment]
@Alovchin91 the problem is not the fact that you mentioned Telegram Desktop, feel free to use whatever client you prefer and to talk about whatever you want here. 
The problem is referring to another app to somehow persuade me to do what you’re asking for…

**Comment by Alovchin91** [comment]
> Would you mind sharing where to get that?

https://github.com/telegramdesktop/tdesktop/issues/25266

**Comment by Alovchin91** [comment]
> The problem is referring to another app to somehow persuade me to do what you’re asking for…

I've explained it in the context of updating through the Store. Besides, I thought you were working with the Telegram folks, so I suppose Telegram Desktop is not a "competing" app. Dunno why it's such a big deal to say "hey, it's still a burden for me, so in your case you probably want to stick with TDesktop instead".

Anyway, I'm outta here.

**Comment by GeldHades27355** [comment]
> > Would you mind sharing where to get that?
> 
> [telegramdesktop/tdesktop#25266](https://github.com/telegramdesktop/tdesktop/issues/25266)

This is unsupported by Telegram. There are no Arm64 release builds of Telegram Desktop available anywhere.
Personally, I don't want to use a secure messenger that is not released or need manual deployment.

Unigram IS officially supported and released on Arm64, so the BY FAR better option on Snapdragon is Unigram Arm64.

Just get it from the store. As I said: you have nothing to lose and only to gain. And the more folks use Arm64 Unigram, the more likely it'll stick around. So why don't you help out by adding your devices to the Arm64 pool?

**Comment by Alovchin91** [comment]
> This is unsupported by Telegram. There are no Arm64 release builds of Telegram Desktop available anywhere.

This post _literally_ says it's an official release:

https://github.com/telegramdesktop/tdesktop/issues/25266#issuecomment-2462801394

> So why don't you help out by adding your devices to the Arm64 pool?

~Don't persuade me to do what you're asking for!~ 😆 /s

**Comment by GeldHades27355** [comment]
> > This is unsupported by Telegram. There are no Arm64 release builds of Telegram Desktop available anywhere.
> 
> This post _literally_ says it's an official release:
> 
> [telegramdesktop/tdesktop#25266 (comment)](https://github.com/telegramdesktop/tdesktop/issues/25266#issuecomment-2462801394)
> 
> > So why don't you help out by adding your devices to the Arm64 pool?
> 
> ~Don't persuade me to do what you're asking for!~ 😆 /s

Thanks! 👍
Interesting tho is that it is NOT available from the Store or the Telegram website. Oh well, might happen yet.

@FrayxRulez and Unigram are usually faster with releases in my experience. 😉

Also: Since Unigram and Telegram share a bunch of code it does make sense that both clients now run as native Arm64.

Still doesn't change my opinion: Just use Unigram for now. You have nothing to lose. And if @FrayxRulez decides to drop Arm64 again switching to Telegram is as simple as installing, signing in and boom - everything comes back. Thanks to the cloud.

**Comment by GeldHades27355** [comment]
Aw hell... It looks like there was an update yesterday (and today) that broke audio playback. 🙄

@FrayxRulez Unigram worked fine until yesterday's update.

Symptoms:
Audio playback takes a long time to start
No sound, although the indicator seems to be progressing
At the end of the message, it stops, but doesn't end playback. It is stuck in the "playing" state

Steps to repro: get a Snapdragon, install Unigram from Store.

Already tried:
Uninstalling and re-installing --> no fix
Installed Telegram Desktop Arm64 (5.7.2) --> works fine
(side note: Telegram Desktop Arm64 consumes 2,5x more RAM than Unigram. 😁)

Sorry team, something's wrong.

**Comment by FrayxRulez** [comment]
@GeldHades27355 Nothing has changed with audio playback

**Comment by GeldHades27355** [comment]
> @GeldHades27355 Nothing has changed with audio playback

Something obviously changed, because it stopped working on three different Snapdragons (SD850, SQ2 and SQ3) simultaneously.

I never said the audio changed, I said audio playback is broken. And since there has been one update yesterday and another today, and it worked fine from Monday till yesterday morning, _something_ broke in one of the last updates.

Actually, this is a worst case scenario, since the store will install the appropriate architecture automatically. So I need to switch to Telegram for now.

How can I help trace this?

Anyone else seeing experiencing this bug?

**Comment by punteroanull** [comment]
Same here on a Surface Pro 11 Elite

Sent from [Proton Mail](https://proton.me/mail/home) for iOS

El vie, 15 nov, 2024 a las 16:13, GeldHades27355 ***@***.***(mailto:El vie, 15 nov, 2024 a las 16:13, GeldHades27355 <<a href=)> ha escrito:

**Comment by FrayxRulez** [comment]
Does it work on 11.4?

**Comment by GeldHades27355** [comment]
> Does it work on 11.4?

Sorry no. Unchanged. Should we open a new issue for better tracking?

PS. small comfort: Telegram Desktop Arm64 also has a bug. 🤣 The "Save as" file picker immediately closes when you try to save media locally. I use Unigram to save media and Telegram to listen to it.

**Comment by FrayxRulez** [comment]
Telegram Desktop bug has already been fixed. 
The only change around the video player has been reverted in 11.4, so everything is exactly as it was in the first arm64 release. Are you sure that the audio has ever worked? I don’t have an arm64 device myself so I’m unable to test (and I have never tried the app myself)

**Comment by GeldHades27355** [comment]
Quite positve. My GF sends me voice messages almost daily, so chances are near 100% that it originally worked. That being said, 11.4 still won't play audio - no change. So now I'm not so sure anymore...

Is it possible this is an issue somewhere else?

Can we help trace this somehow? I'd be happy to support. But I'm not a dev, so I'd need pretty detailed instructions.

(it's telling that the platform you have least incentive to support is already causing problems. 🤣)

**Comment by iamnotafatso** [comment]
The audio playback issue you're experiencing @GeldHades27355 is for voice notes? 

Either way, I've tried sending voice notes and MP3 files, they do play correctly. Running on SD X Plus, Unigram ~~11.3.3~~ 11.4 (maybe it is a processor specific issue?)

**Comment by GeldHades27355** [comment]
> The audio playback issue you're experiencing @GeldHades27355 is for voice notes?
> 
> Either way, I've tried sending voice notes and MP3 files, they do play correctly. Running on SD X Plus, Unigram ~11.3.3~ 11.4 (maybe it is a processor specific issue?)

Correct.
Attached MP3 also won't play.

I doubt that it's about the processor, since Telegram Desktop plays media and voice notes just fine and AFAIK it uses similar / same media libraries as Unigram, no @FrayxRulez ?

Also, I have three Snapdragons here, none of them work as expected:
1. SQ3, Windows 11 Pro dev preview, company managed
2. SQ2, Windows 11 Pro release, company managed
3. SD850, Windows 11 Pro release, personal

Unless there is code specific to the new Elite CPUs, it seems unlikely to me.

**Comment by FrayxRulez** [comment]
@GeldHades27355 No, Telegram Desktop and Unigram don’t have much code in common

**Comment by GeldHades27355** [comment]
> @GeldHades27355 No, Telegram Desktop and Unigram don’t have much code in common

Not even media playback libraries? But obviously, I take your word for it.

In any case, if this were CPU-specific, it'd mean something was specifically made for SD Elite which fails on all other in market SDs.

again @FrayxRulez how can I/we help trace this? Logs? Repros?

**Comment by gus33000** [comment]
I have tried playing voice notes on the following processors I personally own on Unigram 11.4 and I hear them completely fine with no issue, tested list:

- Snapdragon 850
- Snapdragon 8cx Gen 1
- Snapdragon 8cx Gen 3
- Microsoft SQ1
- Microsoft SQ2
- Microsoft SQ3
- Snapdragon X Elite X1E80100
- Snapdragon 7c+ Gen 3

**Comment by gus33000** [comment]
https://github.com/user-attachments/assets/61a41772-731a-4ccf-8f95-3b332d87e6e8

I have also just tested a MP3 and it works just fine as well; see above video for a demonstration

**Comment by GeldHades27355** [comment]
Fun fact: the clip @gus33000 pasted does not play on my SQ3, but works on my SD850. 🤣🤔
(it just chirps about 100ms and stops on the SQ3)

The damndest thing. Here is a short clip of my SD850 showing how Unigram doesn't play, whereas Telegram does.
https://1drv.ms/v/s!AoQI5_3ipPc6pqIY_pVGlpdLd6bIpg?e=vyAyWF
As mentioned: this behavior is IDENTICAL across my SD850, SQ2 and SQ3.

Not sure if you noticed, when I press play on Unigram, the vid's audio gets choppy. So something is happening, just not what is supposed to happen.

My SD850 is as "vanilla" as Snapdragons get. It runs release windows, not M365 joined, has no Audio Processing Objects APOs that could disrupt the audio chain or other oddities I'm aware of.

To be perfectly clear: Unigram and the clip above aside, they play other audio and media just fine.

I'd like to renew my offer to help trace and fix this. What can I do?

**Comment by gus33000** [comment]
This pretty much confirms the issue is not even related to unigram; and not related to this machines but something else on your end. I cant reproduce your issue here, I can also play the video i sent myself fine. So I dont see how its strictly related to unigram. This might be a weird audio configuration on your end. My ARM fleet of devices is not tampered in any way either. Works just fine.

You also claimed previously this didnt happen on older unigram versions and then said it did happen, further enforcing the sentiment that the problem is not unigram related and audio configuration related on your end. Something else changed that is not related to the app; and now that you reproduces the issue in a totally diff app with above video; we can be certain of this.

it is impossible for a bug to both not exist in an old version and exist days after in an old version at once here

![image](https://github.com/user-attachments/assets/4b854722-45df-48f7-8c21-fd25050d8076)

Speaking of "vanilla", i dont think you can get more vanilla than above Snapdragon 850 device

for further completeness; other set of tested devices:

![image](https://github.com/user-attachments/assets/a6496a02-a2c7-47b1-a3f4-45e0ab8df665)

**Comment by gus33000** [comment]
![image](https://github.com/user-attachments/assets/4b854722-45df-48f7-8c21-fd25050d8076)

Speaking of "vanilla", i dont think you can get more vanilla than above Snapdragon 850 device

**Comment by GeldHades27355** [comment]
> You also claimed previously this didnt happen on older unigram versions and then said it did happen, further enforcing the sentiment that the problem is not unigram related and audio configuration related on your end. Something else changed that is not related to the app; and now that you reproduces the issue in a totally diff app with above video; we can be certain of this.
> 
> it is impossible for a bug to both not exist in an old version and exist days after in an old version at once here

Dude - what is your problem? Why are you going all toxic by discrediting what you can clearly see with your own eyes?
Telegram works and Unigram doesn't. On all three devices. It's as simple as that.

Stop claiming there isn't an issue when there is.

And I'm telling you: monday thru wednesday last week Unigram worked just fine. Then three updates were pushed in three days (and more since) and one of them broke audio playback.

Telegram Desktop works fine - **so the issue is with Unigram**.
I'll grant there is another combination of factors we (and especially YOU) don't understand yet.

I offered MULTIPLE times to help trace this - but no one accepted. You specifically are neither helpful nor constructive.

@gus33000 Stop being so arrogant and start suggesting what I can do to to help get to the bottom of this.

@FrayxRulez since Gus isn't helping, how can I get the FIRST release of Unigram from last Monday for testing?
Bear in mind, I'm not a dev. Pulling the code off github and compiling it is beyond me. But I'm perfectly willing to try the executable binary and will provide video proof (or not) of whether the Monday release works.

EDIT: Video playback shows the same symptoms btw. Playback starts with a delay and then only plays the video, but no sound.

**Comment by makisukurisu** [comment]
> Bear in mind, I'm not a dev. Pulling the code off github and compiling it is beyond me. But I'm perfectly willing to try the executable binary and will provide video proof (or not) of whether the Monday release works.

Hi, Fela previously stated that he rolled back all the updates relating to media/audio in one of the updates, but if you insist, you may try to install this version (last pre-release ARM build available to public):
https://t.me/unigraminsiders/523520

Bear in mind that you (most likely) will have to reinstall the app losing all data stored locally (secret chats, cached media, theme settings, etc)

If it does indeed work, could you please also include what is the first version to have the issue to localize the problem as much as possible. (If you already clearly specified it, please restate it)

P.S. if you'll need any assistance installing this older version - get in touch in Unigram Insiders chat, if that's ok with you.

**Comment by gus33000** [comment]
> Dude - what is your problem? Why are you going all toxic by discrediting what you can clearly see with your own eyes? Telegram works and Unigram doesn't. On all three devices. It's as simple as that.

I am not being toxic; I am simply stating that I cannot reproduce an issue and proving that I have the devices in questions; therefore; im adding to this story that its more complicated than you seem to think it is.

> Stop claiming there isn't an issue when there is.

There indeed is no issue on my end; i never said you didnt have an issue?

> And I'm telling you: monday thru wednesday last week Unigram worked just fine. Then three updates were pushed in three days (and more since) and one of them broke audio playback.

Once again; i did not say that was false.

> Telegram Desktop works fine - **so the issue is with Unigram**. I'll grant there is another combination of factors we (and especially YOU) don't understand yet.

There likely is; which is why i added information; and no; the additional setup / env has to be something on your side

> I offered MULTIPLE times to help trace this - but no one accepted. You specifically are neither helpful nor constructive.

Maybe you should consider that we have no idea what to suggest or what you could somehow trace? How would we know?

> @gus33000 Stop being so arrogant and start suggesting what I can do to to help get to the bottom of this.

I am not arrogant; I also do not like the way you are replying to me here. I have spent about 4 hours of my time trying to reproduce this issue to the best I could on multiple devices, a lot of these devices arent even free to begin with. Im deeply sorry to have tried to help here. My bad.

> @FrayxRulez since Gus isn't helping, how can I get the FIRST release of Unigram from last Monday for testing? Bear in mind, I'm not a dev. Pulling the code off github and compiling it is beyond me. But I'm perfectly willing to try the executable binary and will provide video proof (or not) of whether the Monday release works.

Because you think Fela also knows what to "trace"?

So far; you seem to be angry and mad at me for no reason, reasons I do not even understand. But fine. I wont answer further; you seem very aggressive here and unwilling to understand why we somehow dont help you

**Comment by GeldHades27355** [comment]
> > Bear in mind, I'm not a dev. Pulling the code off github and compiling it is beyond me. But I'm perfectly willing to try the executable binary and will provide video proof (or not) of whether the Monday release works.
> 
> Hi, Fela previously stated that he rolled back all the updates relating to media/audio in one of the updates, but if you insist, you may try to install this version (last pre-release ARM build available to public): https://t.me/unigraminsiders/523520
> 
> Bear in mind that you (most likely) will have to reinstall the app losing all data stored locally (secret chats, cached media, theme settings, etc)
> 
> If it does indeed work, could you please also include what is the first version to have the issue to localize the problem as much as possible. (If you already clearly specified it, please restate it)
> 
> P.S. if you'll need any assistance installing this older version - get in touch in Unigram Insiders chat, if that's ok with you.

Thanks for this.

So I just installed 11.3.1 (2) Arm64 Direct.

AND IT WORKS. On all three devices. Immediately and consistently.👍

Happy to share a vid if @gus33000 somehow still thinks my bug report is invalid. 🤪

So now that it is crystal clear that:
1. The original release works
2. Subsequent ones dont
3. It seems that not all changes to the audio playback chain were rolled back

Where does this leave us?

**Comment by gus33000** [comment]
You also seem to have missed the whole point of what I was saying; nowhere did I say your issue didnt exist, I have deduced from my tests and what you said the problem might be related to your audio configuration.

Nowhere did i say i didnt belive you had an issue but it doesnt seem related.

It would gladly help if you shared a bit more information on said audio configuration as a consequence, and stop repeatedly say that; i claimed your issue didnt exist? thats simply false and non productive.

**Comment by gus33000** [comment]
> Happy to share a vid if @gus33000 somehow still thinks my bug report is invalid. 🤪

Between you and me; youre the person that is arrogant here; nowhere did I say this; youre putting words in my mouth and not even reading what Im saying

**Comment by gus33000** [comment]
> So now that it is crystal clear that:

> The original release works
> Subsequent ones dont
> It seems that not all changes to the audio playback chain were rolled back

this is not crystal clear, as I showed you; it works fine for me on a whole slew of devices; and even recorded videos showing so (some contain private information so they got shared to Fela in private on telegram; you can ask him to vouch for that in case you start saying that im saying bullshit somehow; given how toxic you seem to be here towards me)

But it may play a critical role in trying to understand what is different between our two audio setups if and only if you are willing to read what im writing and share a bit more on your audio configuration/setup. (speakers; bluetooth speakers; headphones; bluetooth headphones; aka what your setup is here)

**Comment by FrayxRulez** [comment]
@GeldHades27355 I just went through the 14.500 changes happened between 11.3.1 and 11.4 and there’s nothing related to media playback

**Comment by GeldHades27355** [comment]
Impressive! 👍

So that means it's not directly related to media playback.

Again, I can offer to help trace this and gather more insight if folks point me in the right direction.

@FrayxRulez Which logs, if any, would contain info?

> > So now that it is crystal clear that:
> 
> > The original release works
> > Subsequent ones dont
> > It seems that not all changes to the audio playback chain were rolled back
> 
> this is not crystal clear, as I showed you; it works fine for me on a whole slew of devices; and even recorded videos showing so (some contain private information so they got shared to Fela in private on telegram; you can ask him to vouch for that in case you start saying that im saying bullshit somehow; given how toxic you seem to be here towards me)
> 
> But it may play a critical role in trying to understand what is different between our two audio setups if and only if you are willing to read what im writing and share a bit more on your audio configuration/setup. (speakers; bluetooth speakers; headphones; bluetooth headphones; aka what your setup is here)

As I said multiple times: They are all different in almost every way - except that recent Unigram releases don't play media on any of them. The only commonalities I can see are
- They're all Arm64
- They all run Windows 11
- Unigram after 11.3.1 won't play any media

As I don't know what changed after 11.3.1 (2) Arm64 Direct, there is nothing else I can contribute unless folks have specific requests for me.

Windows event logs don't seem to contain anything meaningful. But maybe I just need to better understand what to look for?

**Comment by gus33000** [comment]
I can reproduce the issue on one device **only** by using a specific pair of BT headphones connected to the device; and even then its quite random. On speakers its reliable as always. Once again; sorry for asking this 3 times but you didnt answer before, can you share more information on your audio topology/configuration?

Thanks Fela as well for the idea that it could be headphones related; definitely seems to play a role here for me.

**Comment by GeldHades27355** [comment]
All devices use their internal speakers only. No BT.

Both Surfaces have default APOs, as im sure you're aware, the SD850 ODM device has none. Direct Qualcomm speaker out. That device uses Qualcomms reference design mobo, which is why it's as vanilla as possible. So is its windows: barely any apps on it, certainly no additional third party drivers. Its release windows 11 24H2 nothing more.

Speaking of which @gus33000 your devices are all up to date on 24H2 as well, right?

**Comment by gus33000** [comment]
Just for context by APOs you likely mean third party ones, qcaud has an apo that's a requirement when designing audio drivers, ProxyAPO.dll but I think I got what you meant either way. Devices are up to date and on latest BSP from Qualcomm (especially the mtp850). The SP11 is not on the production branch, dev external and internal bsp flights but the other devices are on retail tracks. I'm going to work soon I will come back to this later today or during the week but on speakers things work fine here. Will look into headphones more myself with windbg later

**Comment by GeldHades27355** [comment]
Yes I mean third party APOs. QC mandated components for their reference designs (such as our SD850) are beyond our influence.

But are you on 24H2. Not all devices receive it automatically yet. For some, only seekers will get IT.

But even that is irrelevant. The original unigram Arm64 build works on all my devices, the current builds work on none.

Something changed/broke since the original build. @FrayxRulez is the best person to make some educated guesses as to which components could be causing this.

Or, there is a global factor causing all your devices to run current builds and mine only the first one.

Region settings?
Language?
Timezone?
Network?
Local account vs. Online account?
Power management?
System settings?
Windows edition? 
Activation? 

This is intriguing. And finally, we're cooperating @gus33000 👍

**Comment by gus33000** [comment]
I install devices manually as part of dev work they all are on ge_release branches. I am at work at the moment I can't debug using windbg this problem.

26100 ge_release en-us region en-us, mix of both local and msa. Power management is the one set by either the OEM or qc BSP (this is fixed by qcpep bundling the ppkg for it)

**Comment by FrayxRulez** [comment]
@GeldHades27355 Does audio work when playing videos?

**Comment by FrayxRulez** [comment]
By the way, it could really help if, as @makisukurisu mentioned, you could join the Telegram group. In alternative you can write to me directly, same username as here.

**Comment by GeldHades27355** [comment]
Nice to speak with someone else who understands Windows devices.
I'm gonna focus on the SD850, based on the assumption that they'll be most "vanilla" and not bent to hell by OEMs or Surface.

Mine is:
Actually still on 23H2 - build 22631.4460 (sorry, I thought I already brought it to 24H2)
Windows Pro Education edition
Feature Pack 22700.1047.0
OA3 activated
Original image as installed by the ODM
Security Level 1 (standard security)
MSA account enrolled
Language and region: DE-DE
Bluetooth is ON, but no audio devices coupled
WPA3 WiFi
Power management: default profile. No options in this BSP and image.
Unigram is 11.3.1 (2) Arm64 Direct - which (still) works
Vanilla Windows Defender, no 3rd party security

Already tried without success:
AC vs. DC
WiFi vs. offline vs. 4G
DE-DE vs. EN-DE
BT ON vs. OFF
Dev mode ON vs. OFF

Looks like your devices might differ from mine in a few ways:
1. You got dev mode on?
2. Manual OS install vs. OA3 activated and installed by ODM
3. Your SD850 looks like a prototype (EVT or DVT?), mine is a production sample
4. You're on 26xxx builds, mine is still on 22xxx

Ring any bells?

**Comment by GeldHades27355** [comment]
> @GeldHades27355 Does audio work when playing videos?

Yes, but only with Unigram 11.3.1. On all others, it does not. Same symptoms: delayed start, distorted/choppy system audio while Unigram is playing. Unigram gets stuck in the "playing" state at the end of the track and won't change to "stopped" state by itself. Unigram doesn't actually crash. I can't see anything useful in Windows event log - but I'm not entirely sure what to look for either.
(I never learned how to debug during my time at MSFT, I was on the DTS team working with device OEMs so the hardware/firmware side of things)

@FrayxRulez are there debug logs I can turn on somewhere?
and when you mean the "Telegram" group you mean Unigram Preview, right?

**Comment by GeldHades27355** [comment]
I checked Windows' generic audio logs and found this:
Sound level for application [38833FF26BA1D.UnigramPreview_g9c9v27vpyspw!App] changed to [0]
Sound level for application [38833FF26BA1D.UnigramPreview_g9c9v27vpyspw!App] changed to [2]
But all playback apps do that, so it appears expected?

I also get
Stream started.
App Id: 38833FF26BA1D.UnigramPreview_g9c9v27vpyspw!App
PID: 15568
Audio Stream Category: 11

According to EarTrumpet, there is no sound coming from Unigram at all, so it does not appear to be a level/mixer/output issue.

Unless I'm completely wrong, it looks like Unigram is starting system audio playback as it should, but not sending anything.

Codec/streaming issue?

**Comment by FrayxRulez** [comment]
@GeldHades27355 could you please install [this](https://t.me/unigraminsiders/525203)? You'll need to manually install the certificate from file properties into Local Machine > Trusted People. Once the app is installed you need to go to Settings > click 10 times on the app version and set verbosity level to DEBUG.
After that, you can reproduce the issue, go back to the diagnostics menu, set verbosity level to ERROR, click on tdlib_log.txt and send me the logs.

**Comment by FrayxRulez** [comment]
Got a arm64 and found out what's the problem with voice messages. Will be fixed in 11.6!
